src/components/delegate/delegate.component.js

```diff
@@ -13,8 +13,12 @@ const DelegateConstructor = function ($rootScope, $stateParams, $location, $http
         }).then(resp => {
             if (resp.data.success) {
                 vm.address = resp.data;
+
+                if (!vm.address.delegate) {
+                    $state.go('address', { address: $stateParams.delegateId });
+                }
             } else {
-                throw 'Account was not found!';
+                $state.go('home');
```

Comment:
I would much rather get a 404 page in this case. Though it's not the concern of this issue/PR  but something to consider in future.